### PR TITLE
Pact contract testing

### LINKED ISSUE #164: Introduce Pact contract tests
@holly-cummins & I had a [talk at Devoxx Belgium](https://devoxx.be/talk/?id=6661) about Pact contract testing:

- VIdeo replay: https://youtu.be/vYwkDPrzqV8
- Slides: https://speakerdeck.com/hollycummins/how-to-avoid-common-pitfalls-with-modern-microservices-testing-c2a33ee0-ca9d-413e-b01e-cbd25853132a

We used the Quarkus superheroes app as the foundation for our demos. Currently all the code resides in my [own personal fork](https://github.com/edeandrea/quarkus-super-heroes/tree/pact). We need to get this into the main repo.

The code, as-is, will not break anything, but it also will not be executed during any CI/CD process, or dev mode/continuous testing.

There are a few caveats that need to be sorted out.

1. The Pact tests use the [free tier](https://pactflow.io/pricing/) within [Pactflow broker](https://pactflow.io/) - https://quarkus-super-heroes.pactflow.io.
    - There is a CI/CD robot user that has credentials we can store as GitHub secrets, but the larger problem is that the account we currently use is tied to my (@edeandrea's) redhat.com email address. Ideally we should have a more "generic" user to own the account.
    - Under an account you can assign individual users with permissions. @holly-cummins and I are currently in there as admin users.
    - @cescoffier / @maxandersen how do you want to handle this? I can add you both as users in the current Pactflow and you can explore, or I can demo it for you so you get an understanding of it. Let me know your thoughts.
        - You need to authenticate with the broker even to perform read actions
3. The Pact tests as they currently are written will not run by default - they have to have flags to enable them
    - This is because, in its current state, Pact does not work in dev mode/continuous testing. There are some issues on the Pact side and some work on the Quarkus side
    - This way we can safely merge in the code without breaking any of the existing CI/CD workflows and/or dev mode/continuous testing within any of the apps
    - @holly-cummins has already started work on that. In our Devoxx demo we showed it working in continuous testing on the provider side. That was done with a local patch to Pact and the beginnings of a Quarkus extension which @holly-cummins is building
 4. From a CI perspective we should hook in, at a minimum, the provider verification tests into the simple build/test workflow **AND** the nightly CI process that tests against Quarkus `main`
     - The simple build/test workflow runs against all PRs as well as the first workflow against any pushes to `main`.

@holly-cummins did I miss anything?

### PR DESCRIPTION
**Pull request by edeandrea**
Introduce pact contract testing

Closes #164

### CODE REVIEW

**Comment by edeandrea** [comment]
@holly-cummins I updated the commit for the PR based on what we discussed in [this comment](https://github.com/quarkusio/quarkus-super-heroes/issues/164#issuecomment-1284147122).

Would you mind reviewing? I added documentation as well, so please review that too :)

**Comment by edeandrea** [comment]
I also added `<pact_do_not_track>true</pact_do_not_track>` to the surefire setup because I didn't think that https://docs.pact.io/telemetry was something we wanted enabled. Seems its enabled by default and you have to opt out of it.

File: rest-fights/pom.xml [hunk 1]
```diff
@@ -239,6 +246,7 @@
           <systemPropertyVariables>
             <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
             <maven.home>${maven.home}</maven.home>
+            <pact_do_not_track>true</pact_do_not_track>
```
Comment (edeandrea) [inline rest-fights/pom.xml:249]: @holly-cummins I _think_ this is the right thing to do? We don't want to enable https://docs.pact.io/telemetry for the superheroes (or anyone cloning/running superheroes locally)?

File: rest-heroes/pom.xml [hunk 2]
```diff
@@ -166,6 +173,7 @@
           <systemPropertyVariables>
             <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
             <maven.home>${maven.home}</maven.home>
+            <pact_do_not_track>true</pact_do_not_track>
```
Comment (edeandrea) [inline rest-heroes/pom.xml:176]: @holly-cummins see [comment above](https://github.com/quarkusio/quarkus-super-heroes/pull/165/files#r999837297)

File: rest-villains/pom.xml [hunk 3]
```diff
@@ -166,6 +173,7 @@
           <systemPropertyVariables>
             <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
             <maven.home>${maven.home}</maven.home>
+            <pact_do_not_track>true</pact_do_not_track>
```
Comment (edeandrea) [inline rest-villains/pom.xml:176]: @holly-cummins see [comment above](https://github.com/quarkusio/quarkus-super-heroes/pull/165/files#r999837297)

**Review by holly-cummins — APPROVED** [review]

**Pull request merged**